test/compress/dead-code.js

```diff
@@ -230,3 +230,25 @@ accessor: {
     }
     expect: {}
 }
+
+issue_2233: {
+    options = {
+        pure_getters: "strict",
+        side_effects: true,
+        unsafe: true,
+    }
+    input: {
+        Array.isArray;
+        Boolean;
+        console.log;
+        Error.name;
+        Function.length;
+        Math.random;
+        Number.isNaN;
+        RegExp;
+        Object.defineProperty;
+        String.fromCharCode;
```

Comment:
Could you please add:
```js
    UndeclaredGlobal;
    function foo() {
        Math.sin;
        AnotherUndeclaredGlobal;
        Number;
    }
```
and just to see what happens in a different test - redeclare a well known builtin used above with "reduce_vars" and "unused" enabled:
```js
    var Boolean;
```